lib/logstash/codecs/auto_flush.rb

```diff
@@ -7,20 +7,48 @@ def initialize(mc, interval)
     @stopped = Concurrent::AtomicBoolean.new # false by default
   end
 
+  # def start
+  #   # can't start if pipeline is stopping
+  #   return self if stopped?
+  #   if pending?
+  #     @task.cancel
+  #     create_task
+  #   elsif finished?
+  #     create_task
+  #   # else the task is executing
+  #   end
+  #   self
+  # end
+
   def start
     # can't start if pipeline is stopping
     return self if stopped?
-    if pending?
-      @task.reset
-    elsif finished?
-      @task = Concurrent::ScheduledTask.execute(@interval) do
-        @mc.auto_flush()
-      end
-    # else the task is executing
+
+    if pending? && @task.cancel
+      create_task
+      return
+    end
+    # maybe we have a timing edge case
+    # where pending? was true but cancel failed
+    # because the task started running
+    if finished?
+      create_task
+      return
     end
+    # else the task is executing
+    # wait for task to complete
+    # flush could feasibly block on queue access
+    @task.value
+    create_task
     self
```

Comment:
Sorry - I did not understand what you meant.

Yes of course - all short circuit returns should return self.